### ISSUE TITLE
Training loss graph is updated when "discretize output" is selected or deselected

### ISSUE THREAD

**Issue by tobimag** [tensorflow/playground#184]
The graph visualizing the training loss is updated every time the "discretize output" box is selected or deselected, please see attached screen recording. Because no step is actually performed this is assumed to be a bug. 

https://user-images.githubusercontent.com/5433242/227735348-754e36d2-727e-45f8-add5-f9591a4fe497.mov

**Comment by kandelak** [comment]
I agree that this is clearly a bug. Nevertheless, I wanted to report another issue related to this one. If you click the Discetize output, it does have to update the training loss numerical value and graph, but just once. Whenever you discretize the space and neglect the existence of the"gradient" colours, you only have 2 static values for the predicted values, hence calculating the loss would result in a different result.
